oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/session/OidcProviderSessionRegistry.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.oidc.authentication.session;
+
+import org.springframework.security.oauth2.client.oidc.authentication.logout.OidcLogoutToken;
+
+/**
+ * A registry to record the tie between the OIDC Provider session and the Client session.
+ * This is handy when a provider makes a logout request that indicates the OIDC Provider
+ * session or the End User.
+ *
+ * @author Josh Cummings
+ * @since 6.1
+ * @see <a target="_blank" href=
+ * "https://openid.net/specs/openid-connect-backchannel-1_0.html#LogoutToken">Logout
+ * Token</a>
+ */
+public interface OidcProviderSessionRegistry {
+
+	/**
+	 * Register a OIDC Provider session with the provided client session. Generally
+	 * speaking, the client session should be the session tied to the current login.
+	 * @param details the {@link OidcProviderSessionRegistrationDetails} to use
+	 */
+	void register(OidcProviderSessionRegistrationDetails details);
+
+	/**
+	 * Update the entry for a Client when their session id changes. This is handy, for
+	 * example, when the id changes for session fixation protection.
+	 * @param oldClientSessionId the Client's old session id
+	 * @param newClientSessionId the Client's new session id
+	 */
+	void reregister(String oldClientSessionId, String newClientSessionId);
```

Comment:
Can we think of another name  instead of `reregister`.

Reply:
Personally, I like `reregister`. Could you explain why you don't? What I like about it is that it has the word `register` in it, which aligns well with the class name and the related method names. It's also a verb like the other two:

* `register` - add a registration
* `reregister` - replace a registration
* `deregister` - remove a registration

I also like that `register`/`reregister`/`deregister` are more descriptive than `add`/`replace`/`remove` while minimizing  repetition of the class name.

I'm open to discussing names; my main point is that we should consider how the three method names go together before changing.